engine/equations.cc

```diff
@@ -641,9 +641,16 @@ namespace MathDAG
     shared_ptr<VariableDAG> r(new VariableDAG(valueId, nm, type));
     expressionCache.insert(valueId, r);
     r->init=vv->init;
-    if (auto v=minsky.definingVar(valueId))
+    if (auto v=minsky.definingVar(valueId)) {
       if (v->type()!=VariableType::integral && v->numPorts()>1 && !v->ports[1]->wires().empty())
         r->rhs=getNodeFromWire(*v->ports[1]->wires()[0]);
+	  // ensure that flows used as intial conditions, which inherit their values from other vars, correctly initialise corresponding stock var. for ticket 1137.
+	  auto vi=minsky.variableValues[VariableValue::valueId(v->group.lock(),vv->init)];
```

Comment:
vv->init may be an expression, like 2 foobar, where foobar is the name of another variable. In this case, vi is of type undefined.

Reply:
Thanks, I realized that when I managed to get the init value field of the int op and Godley stock var in Kresimir's examples to be populated by the correct name but not its value.